iothub_client/src/iothub_client_ll_uploadtoblob.c

```diff
@@ -1064,7 +1105,50 @@ IOTHUB_CLIENT_RESULT IoTHubClient_LL_UploadToBlob_SetOption(IOTHUB_CLIENT_LL_UPL
                 }
             }
         }
-        else
+		/*Codes_SRS_IOTHUBCLIENT_LL_32_008: [ OPTION_HTTP_PROXY - then the value will be a pointer to HTTP_PROXY_OPTIONS structure. ]*/
+		else if (strcmp(optionName, OPTION_HTTP_PROXY) == 0)
+		{
+			HTTP_PROXY_OPTIONS* proxy_options = (HTTP_PROXY_OPTIONS *)value;
+			
+			if (proxy_options->host_address == NULL)
+			{
+				/* Codes_SRS_IOTHUBCLIENT_LL_32_006: [ If `host_address` is NULL, `IoTHubClient_LL_UploadToBlob_SetOption` shall fail and return `IOTHUB_CLIENT_INVALID_ARG`. ]*/
+				LogError("NULL host_address in proxy options");
+				result = IOTHUB_CLIENT_INVALID_ARG;
+			}
+			/* Codes_SRS_IOTHUBCLIENT_LL_32_007: [ If only one of `username` and `password` is NULL, `IoTHubClient_LL_UploadToBlob_SetOption` shall fail and return `IOTHUB_CLIENT_INVALID_ARG`. ]*/
+			else if (((proxy_options->username == NULL) || (proxy_options->password == NULL)) &&
+				(proxy_options->username != proxy_options->password))
+			{
+				LogError("Only one of username and password for proxy settings was NULL");
+				result = IOTHUB_CLIENT_INVALID_ARG;
+			}
+			else
+			{
+				handleData->http_proxy_port = proxy_options->port;
+
+				if (mallocAndStrcpy_s(&handleData->http_proxy_hostname, proxy_options->host_address) != 0)
```

Comment:
>handleData->http_proxy_hostname [](start = 27, length = 31)

You need to free these first if they are already initialize from a previous call here.